### PR TITLE
Add routine for better spawning of high command vehicles

### LINKED ISSUE #2999: [Feature]: Write a better high command remote vehicle spawn
Currently when you buy a high command squad when not near HQ, it uses findEmptyPosition to find a random "empty" spot near the HQ marker. This works poorly: Vehicles are sometimes "safe" but incapable of leaving HQ, and with the new building placer, frequently explode on spawn.

Placing on roads works for avoiding immediate explosions, but has the issue in this case that the travel direction is unknown at spawn, and Arma AI is *really* bad at turning. Thinking of writing a new placer algorithm that looks for places just off-road and perpendicular to it.

### PR DESCRIPTION
**Pull request by jaj22**
## What type of PR is this.
1. [x] Bug
2. [ ] Change
3. [X] Enhancement

### What have you changed and why?
With the introduction of the building placer, remotely recruiting vehicles for high command became even less reliable. This is a new routine that attempts to find empty spots facing onto roads, which does a much better job of picking a place where vehicles can actually drive out of HQ in various directions. Deals ok with various obstacle types and terrain gradients.

Fallback uses the old random placement but with a larger radius.

### Please specify which Issue this PR Resolves.
closes #2999

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

File: A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+private _vehicle = createVehicle [_vehType, [0,0,-1000], [], 0, "CAN_COLLIDE"];
+_vehicle enableSimulation false;
+
+private _bb = boundingBoxReal _vehicle;
+private _bbSize = _bb#1 vectorDiff _bb#0;
+_bbSize params ["_vehWidth", "_vehLength", "_vehHeight"];
+private _slotSize = _bbSize vectorAdd [1.5, 0.3, 0.2];
+_slotSize params ["_slotWidth", "_slotLength", "_slotHeight"];
+private _slotRad = sqrt (_slotWidth^2 + _slotLength^2);
```
Comment (killerswin2) [inline A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf:29]: ugggghhhh, why doesn't the hypot command exist

**Review by killerswin2 — APPROVED** [review]
LGTM

File: A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    5. <array> Non-empty to create AA truck
+*/
+
+#include "..\..\script_component.hpp"
+FIX_LINE_NUMBERS()
+
+// Same params as HR_GRG_fnc_confirmPlacement but doesn't use the callbacks
+params ["_vehType", "", "", "_groupParams", "_mounts"];
+
+private _vehicle = createVehicle [_vehType, [0,0,-1000], [], 0, "CAN_COLLIDE"];
```
[the remark this answers is not in the document]
Comment (jaj22) [reply A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf:21]: huh. Really? Even with CAN_COLLIDE?

File: A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf (hunk 1 above)
Comment (jaj22) [reply A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf:29]: could have done `vectorMagnitude [_slotWidth, _slotLength]`

Comment (killerswin2) [reply A3A/addons/core/functions/REINF/fn_spawnHCVeh.sqf:29]: nah, that's slow. Arma has to create that array

**Pull request merged**